src/nprintml/cli.py

```diff
@@ -35,14 +37,14 @@ def execute(argv=None, **parser_kwargs):
 
         check_output_directory(args)
 
-        for (step, results) in pipeline(args):
+        for (step, results) in pipeline(parser, args):
             print(step, results, sep=' → ')
 
         print('done →', pipeline.results)
     except KeyboardInterrupt:
         print('interrupted ✕')
     except Exception as exc:
-        if args.traceback:
+        if args is not None and args.traceback:
```

Comment:
This is unrelated, actually, but I smuggled it in :sweat_smile: 

In the unlikely case that there's an exception in `build_parser` or otherwise before we've parsed the CLI arguments, then this reference to `args` was undefined, and raising its own, obfuscating exception.